stat/stat.go

```diff
@@ -217,6 +217,59 @@ func Correlation(x, y, weights []float64) float64 {
 	return (sxy - xcompensation*ycompensation/sumWeights) / math.Sqrt(sxx*syy)
 }
 
+// KendallCorrelation returns the weighted Tau-a Kendall correlation between the
+// samples of x and y.
+// The lengths of x and y must be equal. If weights is nil then all of the
+// weights are 1. If weights is not nil, then len(x) must equal len(weights).
+func KendallCorrelation(x, y, weights []float64) float64 {
+	if len(x) != len(y) {
+		panic("stat: slice length mismatch")
+	}
+
+	var (
+		cc float64 // number of concordant pairs
+		dc float64 // number of discordant pairs
+		n  = len(x)
+	)
+
+	if weights == nil {
+		for i := 0; i < n; i++ {
+			for j := i; j < n; j++ {
+				if i == j {
+					continue
+				}
+				if math.Signbit(x[j]-x[i]) == math.Signbit(y[j]-y[i]) {
+					cc++
+				} else {
+					dc++
+				}
+			}
+		}
+		return (cc - dc) / float64(n*(n-1)/2)
+	}
+
+	var sumWeights float64
+
+	var c int
+
+	for i := 0; i < n; i++ {
+		for j := i; j < n; j++ {
+			if i == j {
+				continue
+			}
+			c++
+			var weight = weights[i] * weights[j]
```

Comment:
`weight := `

Reply:
Is this a style issue?

Reply:
Yes.

Reply:
Given the life-span of the temporary, I'd even go so far as to say `w := weights[i] * weights[j]`, but it's not necessary to change that now.